djactasauth/views.py

```diff
@@ -1,41 +1,15 @@
-import django
-from django.contrib.auth.forms import AuthenticationForm
-from django.views.decorators.debug import sensitive_post_parameters
-from djactasauth.forms import InitialValuesFromRequestGetFormMixin
+from django.contrib.auth.views import LoginView
 
 
-def get_login_form(request, authentication_form=AuthenticationForm, **kwargs):
-    base_form = authentication_form
-    if django.VERSION[:2] < (1, 6):
-        class CaptureRequestActAsAuthForm(base_form):
-            def __init__(self, *a, **kw):
-                if request.method == 'POST':
-                    kw.setdefault('request', request)  # "backport" from 1.6
-                super(CaptureRequestActAsAuthForm, self).__init__(*a, **kw)
+class PrefillLoginView(LoginView):
 
-        base_form = CaptureRequestActAsAuthForm
+    query2initial = ('username',)
 
-    if not issubclass(base_form, InitialValuesFromRequestGetFormMixin):
-        class InitialFromQueryAuthForm(
-                InitialValuesFromRequestGetFormMixin, base_form):
-
-            @property
-            def query2initial(self):
-                return super(InitialFromQueryAuthForm, self).query2initial + \
-                        ('username',)
-
-        base_form = InitialFromQueryAuthForm
-    else:
-        # TODO: raise warning if it doesn't have username in query2initial
-        pass
-
-    return base_form
-
-
-@sensitive_post_parameters('password')
-def act_as_login_view(request, **kwargs):
-    # in-mehod import otherwise cannot mock it due to sensitive wraps
-    # see test_act_as_login_view_protects_sensitive_vars
-    from django.contrib.auth.views import login
-    kwargs['authentication_form'] = get_login_form(request, **kwargs)
-    return login(request, **kwargs)
+    def get_initial(self):
+        initial = super(PrefillLoginView, self).get_initial()
+        field_names = set(self.query2initial)
+        for field_name in field_names:
```

Comment:
@Eorhim In what use case would one have more than one field here?

Reply:
@belugame If you create a more elaborate login form (think about an additional "company" field), then you could fill it with adding it to `self.query2initial`.